src/ewatercycle/observation/grdc.py

```diff
@@ -98,30 +99,136 @@ def get_grdc_data(
     if not data_path.exists():
         raise ValueError(f"The grdc directory {data_path} does not exist!")
 
-    # Read the raw data
-    raw_file = data_path / f"{station_id}_{parameter}_Day.Cmd.txt"
+    # Read the NetCDF file
+    nc_file = data_path / "GRDC-Daily.nc"
+    station_not_in_nc = False
+    if nc_file.exists():
+        ds = xr.open_dataset(nc_file)
+        try:
```

Comment:
```py
if not int(station_id) in ds["id"]:
```
should work here. Would avoid issues with the time selection or renaming making it appear like the station is not in the netcdf file.